src/cargo/util/paths.rs

```diff
@@ -129,6 +158,12 @@ pub fn append(path: &Path, contents: &[u8]) -> CargoResult<()> {
     Ok(())
 }
 
+pub fn mtime(path: &Path) -> CargoResult<FileTime> {
+    let meta =
+        fs::metadata(path).chain_err(|| internal(format!("failed to stat `{}`", path.display())))?;
```

Comment:
In general `internal` hasn't worked out great for Cargo and I've been trying to remove it wherever I see it, so while you're here mind removing this `internal`?